core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -751,7 +751,7 @@ private[spark] object Utils
       throw new IOException(s"Failed to create directory ${targetDir.getPath}")
     }
     val dest = new File(targetDir, filename.getOrElse(path.getName))
-    if (fs.getFileStatus(path).isFile) {
+    if (SparkHadoopUtil.isFile(fs, path)) {
```

Comment:
Are there some issues with the original logic in some places? It might need to be carefully considered. For instance, here, if the `path` does not exist, it would enter the else branch, and then a `FileNotFoundException` would be thrown. Although the result hasn't changed, perhaps the `FileNotFoundException` should be thrown earlier.